sdk/keyvault/keyvault-admin/src/backupClient.ts

```diff
@@ -0,0 +1,277 @@
+      ...{
+        loggingOptions: {
+          logger: logger.info,
+          logPolicyOptions: {
+            allowedHeaderNames: [
+              "x-ms-keyvault-region",
+              "x-ms-keyvault-network-info",
+              "x-ms-keyvault-service-version"
+            ]
+          }
+        }
+      }
+    };
+
+    const pipeline = createPipelineFromOptions(internalPipelineOptions, authPolicy);
+    this.client = new KeyVaultClient({
+      apiVersion: pipelineOptions.serviceVersion || LATEST_API_VERSION,
+      ...pipeline
+    });
+  }
+
+  /**
+   * Starts generating a backup of an Azure Key Vault on the specified Storage Blob account.
+   *
+   * This function returns a Long Running Operation poller that allows you to wait indefinitely until the Key Vault backup is generated.
+   *
+   * Example usage:
+   * ```ts
+   * const client = new KeyVaultBackupClient(url, credentials);
+   *
+   * const blobStorageUri = "<blob-storage-uri>"; // <Blob storage URL>/<folder name>
+   * const sasToken = "<sas-token>";
+   * const poller = await client.beginBackup(blobStorageUri, sasToken);
+   *
+   * // Serializing the poller
+   * const serialized = poller.toString();
+   * // A new poller can be created with:
+   * // await client.beginBackup(blobStorageUri, sasToken, { resumeFrom: serialized });
+   *
+   * // Waiting until it's done
+   * const backupUri = await poller.pollUntilDone();
+   * console.log(backupUri);
+   * ```
+   * @summary Creates a new role assignment.
+   * @param blobStorageUri The URL of the blob storage resource, with the path to the folder name where the backup will end up being generated.
+   * @param sasToken The SAS token.
+   * @param [options] The optional parameters.
+   */
+  public async beginBackup(
+    blobStorageUri: string,
+    sasToken: string,
+    options: BeginBackupOptions = {}
+  ): Promise<PollerLike<PollOperationState<string>, string>> {
+    if (!(blobStorageUri && sasToken)) {
+      throw new Error(
+        "beginBackup requires non-empty strings for the parameters: blobStorageUri and sasToken."
+      );
+    }
+
+    const poller = new BackupPoller({
+      blobStorageUri,
+      sasToken,
+      client: this.client,
+      vaultUrl: this.vaultUrl,
+      intervalInMs: options.intervalInMs,
+      resumeFrom: options.resumeFrom,
+      requestOptions: options
+    });
+
+    // This will initialize the poller's operation (the generation of the backup).
+    await poller.poll();
+
+    return poller;
+  }
+
+  /**
+   * Starts restoring all key materials using the SAS token pointing to a previously stored Azure Blob storage
+   * backup folder.
+   *
+   * This function returns a Long Running Operation poller that allows you to wait indefinitely until the Key Vault backup is generated.
+   *
+   * Example usage:
+   * ```ts
+   * const client = new KeyVaultBackupClient(url, credentials);
+   *
+   * const blobStorageUri = "<blob-storage-uri>"; // <Blob storage URL>/<folder name>
```

Comment:
In this case the folder name is a separate argument.
```suggestion
   * const blobStorageUri = "<blob-storage-uri>"; 
```

Reply:
@christothes I thought the same, but my tests pass with the same parameters I'm using in the others, which satisfy: `<Blob storage URL>/<folder name>`. Do you have different parameters for these tests?

Reply:
Oh, you are correct, sorry!